src/ast.h

```diff
@@ -140,6 +144,8 @@ struct FuncDeclaration {
 };
 
 struct Func {
+  Func();
+
```

Comment:
I see that you have mulltiple cases where a Func instance is created with a name, or a name and flags. Why not define constructors for those cases so that you don't duplicate code to initiate such cases?

Again, I don't know if you expect to do that as a separate step.